### ISSUE TITLE
Documentation Needed!

### ISSUE THREAD

**Issue by peymayesh** [badoo/Chatto#146]
This project is great, but it's too confusing because of nested usage of `Generics`.
I'm trying to create a simple `customMessageView` for **one week**! It took just two days for me to get up & running with `AsyncDisplayKit` because of it's good **DOCUMENTATION**.
At least, please provide me a **hierarchy** of `classes` and `protocols` so I know how to make it done right.

Thanks! 🙏

**Comment by diegosanchezr** [comment]
Hi @peymayesh,

Thank you for your feedback. We do plan to improve the documentation for the next release, which should happen "soon". 

Could you elaborate a little bit more about what was the most difficult part for you?

**Comment by peymayesh** [comment]
Hi @diegosanchezr,
I wanted to create a Photo+Text Bubble View, so I read ChattoApp and rewrite it for my models, but there is too many nested protocols and classes that I failed every time.
For now I'm back on AsyncDisplayKit, but I hope you make an easy-understandable example or complete the documentation so that I migrate to this great project.

Thanks for response. 👍

**Comment by makoni** [comment]
In-code documentation + jazzy generated documentation would be great

http://www.appcoda.com/swift-markdown/

**Comment by Axcic** [comment]
On 5th June you've said it'll come in next release. But it's not(

**Comment by diegosanchezr** [comment]
@Axcic Please take a look at https://github.com/badoo/Chatto/wiki

We made a big effort on that, I know there are things missing and room for improvement. Unfortunately we're short on resources to keep working on that but we do accept contributions.

**Comment by apwelsh** [comment]
Any plans to revisit the documentation?  I have been out of the loop on Swift long enough, that just having the code is not enough.  As stated in the OP's original posting, the multiple levels of abstraction are extremely hard to figure out just how all this is strung together.  The Demo project is not enough.  I don't wish to just blindly copy a project, and then refactor it to make it work.  I need to understand how it works.  Would there be any chance of getting a super simple -- as simple as possible -- example of creating the most basic chat app that can handle text and image data?  Do we really need to build a complete library of implementation classes to handle every little piece of this, or do the ChattoAdditions provided text and image classes have a simpler solution?  I am really lost here.  I was using JSQMessagesViewController and had no issue learning it.  I then switched to QMChatViewController, but it is limited to requiring I provide a URI for image attachments, so I found this.  It has the look and feel I want, and appears to work with iPhone X flawlessly.  I just cannot for the life of me figure out how all the puzzle pieces fit together.  Just a simply diagram of the classes, and protocols, and how they are related.  And maybe how the various functions rely on them would really be enough to get started.  But also, finishing up the tutorial to actually construct a working controller would be helpful too.  The current one leave too much out.
